index.js

```diff
@@ -96,7 +96,7 @@ class TuyaLan {
             } catch(ex) {}
 
             //if (!/^[0-9a-f]+$/i.test(device.id)) return this.log.error('%s, id for %s, is not a valid id.', device.id, device.name || 'unnamed device');
-            if (!/^[0-9a-f]+$/i.test(device.key)) return this.log.error('%s, key for %s (%s), is not a valid key.', device.key.replace(/.{4}$/, '****'), device.name || 'unnamed device', device.id);
+            if (!/^[a-zA-Z0-9~`!@#$%^&*()+={}\[\];:\'\"<>.,\/\\\?-_]+$/i.test(device.key)) return this.log.error('%s, key for %s (%s), is not a valid key.', device.key.replace(/.{4}$/, '****'), device.name || 'unnamed device', device.id);

```

Comment:
I think the dash ('-') needs to be at the end else is interpreted as a range modifier (like the dash in a-z).

Reply:
good catch, fixed